### PR TITLE
mimic: librados: Invalidate crc in reused read buffer

### PR DESCRIPTION
**Pull request by badone**
Fixes: https://tracker.ceph.com/issues/40765

Signed-off-by: Brad Hubbard <bhubbard@redhat.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/librados/IoCtxImpl.cc [hunk 1]
```diff
@@ -742,6 +742,11 @@ int librados::IoCtxImpl::operate_read(const object_t& oid,
 
   Context *onack = new C_SafeCond(&mylock, &cond, &done, &r);
 
+  // invalidate any existing crc (if the buffer is reused)
+  if (pbl) {
```
Comment (jdurgin) [inline src/librados/IoCtxImpl.cc]: would aio_operate_read() have the same problem?

**Comment by jdurgin** [comment]
is the bug present in master too? if so it would make sense to fix it there, then backport to n/m/l assuming they're all affected

File: src/librados/IoCtxImpl.cc (hunk 1 above)
Comment (badone) [reply src/librados/IoCtxImpl.cc]: Looking into this now.

**Comment by badone** [comment]
Hi @jdurgin I double-checked master and nautilus using v2 and v1 and neither appears to be affected. I'll see if I can work out why this is so we can be absolutely confident.

**Comment by gregsfortytwo** [comment]
Ping @badone did you do that? We'd like to close the associated bug. :)

**Comment by badone** [comment]
@gregsfortytwo Sorry, I'll see if I can get it finished today.

**Comment by badone** [comment]
Closing this since this should have been resolved by backporting https://tracker.ceph.com/issues/38416

**Pull request closed without merging**